From 6a334f57b6dd21a522258941fdd90d20c4d7197d Mon Sep 17 00:00:00 2001
From: Nicolas Boulenguez <nicolas@debian.org>
Date: Thu, 25 Jul 2024 10:50:43 +0200
Subject: [PATCH 2/2] scripts/mk: fix #1076863 DEB_VERSION_UPSTREAM_REVISION
 but keep lazy expansion

In e146a68a, the variables (list A) are assigned with a lazy expansion
wrapper, so that first expansion triggers dpkg-parsechangelog and
actually assigns the variables (list B) with their final values.

A mispelling of DEB_VERSION_UPSTREAM_REVISION in (A) causes #1076863
and its clones (thanks to Michael Tokarev <mjt@tls.msk.ru>).

The issue was not detected by tests and hard to understand because (B)
was correct, hiding the bug after any other expansion.  For example,
variables expansions within recipes were less likely to be affected
because they happen long after plain assignments.

Testing this for regression would require a separate test for each
variable and does not seem worth the while.

Commit c05f8297 fixes the bug but runs dpkg-parsechangelog each time a
debian/rules including pkg-info.mk is parsed.

This fix preserves lazy expansion.
---
 scripts/mk/pkg-info.mk | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/scripts/mk/pkg-info.mk b/scripts/mk/pkg-info.mk
index ddda4f736..a51dd88e4 100644
--- a/scripts/mk/pkg-info.mk
+++ b/scripts/mk/pkg-info.mk
@@ -35,7 +35,7 @@ ifdef SOURCE_DATE_EPOCH
   $(call dpkg_lazy_eval,DEB_VERSION,$$(dpkg_parsechangelog_run))
   $(call dpkg_lazy_eval,DEB_VERSION_EPOCH_UPSTREAM,$$(dpkg_parsechangelog_run))
   $(call dpkg_lazy_eval,DEB_VERSION_UPSTREAM,$$(dpkg_parsechangelog_run))
-  $(call dpkg_lazy_eval,DEB_UPSTREAM_REVISION,$$(dpkg_parsechangelog_run))
+  $(call dpkg_lazy_eval,DEB_VERSION_UPSTREAM_REVISION,$$(dpkg_parsechangelog_run))
 else
   # We need to compute the values right now.
   $(dpkg_parsechangelog_run)
-- 
2.39.2

